hyperdrive/packages/signer/sign/src/lib.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::{anyhow, Result};
+use hyperware_process_lib::net::{NetAction, NetResponse};
+use hyperware_process_lib::{await_message, call_init, Address, Message, Request, Response};
+use serde::{Deserialize, Serialize};
+
+wit_bindgen::generate!({
+    path: "target/wit",
+    world: "sign-sys-v0",
+    generate_unused_types: true,
+    additional_derives: [PartialEq, serde::Deserialize, serde::Serialize, process_macros::SerdeJsonInto],
+});
+
+#[derive(Debug, Serialize, Deserialize)]
+enum SignerRequest {
+    Sign(SignRequest),
+    Verify { from: Address, data: SignResponse },
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct SignRequest {
+    pub site: String,
+    pub time: u64,
+    pub nonce: Option<String>,
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct SignResponse {
+    pub body: SignRequest,
+    pub message: Vec<u8>,
+    pub signature: Vec<u8>,
+}
```

Comment:
Move these types into the `login:sys-v0.wit` file. This may require a slight change in the types (e.g. you can have an enum similar to the above but there is no `{ }` syntax; you need to use `( )` to supply args).

See, e.g., https://github.com/hyperware-ai/hyperdrive/blob/main/hyperdrive/packages/homepage/api/homepage%3Asys-v1.wit

See also this specific subsection, which defines a few conventions we use for package WIT APIs: https://book.hyperware.ai/cookbook/package_apis.html#exporting-an-api

In particular, the `interface` should have name corresponding to the process